File: app/src/main/java/com/github/premnirmal/ticker/Injector.java
package com.github.premnirmal.ticker;

import dagger.ObjectGraph;

/**
 * Created by premnirmal on 2/20/15.
 */
public class Injector {

    private ObjectGraph objectGraph;

    private static Injector instance;

    static void init(StocksApp app) {
        instance = new Injector(app);
    }

    private Injector(StocksApp app) {
        Tools.init(app);
        objectGraph = ObjectGraph.create(new AppModule(app));
        instance = this;
    }

    public static <T> T get(Class<T> clazz) {
        return instance.objectGraph.get(clazz);
    }

    public static void inject(Object object) {
        instance.objectGraph.inject(object);
    }

}

File: app/src/main/kotlin/com/github/premnirmal/ticker/components/Injector.kt
package com.github.premnirmal.ticker.components

import com.github.premnirmal.ticker.StocksApp
import com.github.premnirmal.ticker.ui.AppMessaging
import dagger.hilt.EntryPoints

/**
 * Created by premnirmal on 2/26/16.
 */
object Injector {

  private lateinit var app: StocksApp

  fun init(app: StocksApp) {
    this.app = app
    AppMessaging.context = app
  }

  fun appComponent(): AppEntryPoint {
    return EntryPoints.get(app, AppEntryPoint::class.java)
  }
}